docs/diagnostic-logs.md

```diff
@@ -176,28 +259,99 @@ Service reloading, please reconnect | Azure SignalR Service is reloading. Azure
 
+#### Message related issues
+
+When encountering message related problem, you can take advantage of messaging logs to troubleshoot. Firstly, [enable diagnostic logs](#enable-diagnostic-logs) in service, logs for server and client.
+
+> For ASP.NET Core, see [here](https://docs.microsoft.com/aspnet/core/signalr/diagnostics) to enable logging in server and client.
+> 
+> For ASP.NET, see [here](https://docs.microsoft.com/aspnet/signalr/overview/testing-and-debugging/enabling-signalr-tracing) to enable logging in server and client.  
+
+If you don't mind potential performance impact and no client-to-server direction message, check the `Messaging` in `Log Source Settings/Types` to enable *collect-all* log collecting behavior. For more information about this behavior, see [collect all section](#collect-all). 
+
+Otherwise, uncheck the `Messaging` to enable *collect-partially* log collecting behavior. This behavior requires configuration in client and server to enable it. For more information, see [collect partially section](#collect-partially).
+
+##### Message loss
+
+If you encounter message loss problem, the key is to locate the place where you lose the message. Basically, you have 3 components when using SignalR service: SignalR service, server and client. Both server and client are connected to SignalR service, they don't connected to each other directly once negotiation is completed. Therefore, we need to consider 2 directions for messages, for each direction, we need to consider 2 paths:
+
+* From client to server via SignalR service
+  * Path 1: Client to SignalR service
+  * Path 2: SignalR service to server
+* From server to client via SignalR service
+  * Path 3: Server to SignalR service 
+  * Path 4: SignalR service to client
+
+![Message path](./images/diagnostic-logs/message-path.png)
+
+For **collect all** collecting behavior:
+
+SignalR service only trace messages in direction **from server to client via SignalR service**. The tracing ID will be generated in server, the message will carry the tracing ID to SignalR service.
+
+> If you want to trace message and [send messages from outside a hub](https://docs.microsoft.com/en-us/aspnet/core/signalr/hubcontext) in your app server, you need to enable **collect all** collecting behavior to collect message logs for the messages which are not originated from diagnostic clients.
```

Comment:
so there is no way to collect client to server message and message sent outside hub method at the same time?

Reply:
Customer can turn on collect-all and mark the client as diagnostic client at the same time. 

As we discussed offline,
1. We can make the message invoked outside of hub class has a way to get logged without turning on collect-all in the future.
2. Collect logs of client to server message as advanced option

Reply:
> Customer can turn on collect-all and mark the client as diagnostic client at the same time.

This is not clear by reading the doc. I think customer may get confused that diagnostic client only works when he chooses to collect partially.

Reply:
I will add a new line to explain, and add a link to the diagnostic section

> Diagnostic clients works for both **collect all** and **collect partially** collecting behaviors. It has higher priority to collect logs. For more information, see [diagnostic client section](#diagnostic-client).